### PR TITLE
Add reversed(torch.Tensor)

### LINKED ISSUE #3376: [minor] support reversed(torch.Tensor)
```python
sorted(torch.Tensor([3, 2, 1])) # works

reversed(torch.Tensor([3, 2, 1])) # doesn't 
#Traceback (most recent call last):
#  File "<stdin>", line 1, in <module>
#TypeError: argument to reversed() must be a sequence

reversed(np.array([3, 2, 1])) # works
#<reversed object at 0x7f2ea830b290>
```

This can be addressed by implementing [`__reversed__`](https://bugs.python.org/issue18826) magic.

(Workaround, of course, is `reversed(list(torch.Tensor([3, 2, 1])))`)

### PR DESCRIPTION
**Pull request by vishwakftw**
Closes https://github.com/pytorch/pytorch/issues/3376

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -6194,6 +6194,16 @@ def _test_flip(self, use_cuda=False):
     def test_flip(self):
         self._test_flip(self, use_cuda=False)
 
+    def test_reversed(self):
+        val = torch.arange(0, 10)
+        self.assertEqual(reversed(val), torch.arange(9, -1, -1))
+
+        val = torch.arange(0, 10).view(3, 3)
```
Comment (SsnL) [inline test/test_torch.py]: should be `arange(1, 10)`

Comment (vishwakftw) [reply test/test_torch.py]: Sorry about that. Will fix.

File: test/test_torch.py [hunk 2]
```diff
@@ -6194,6 +6194,16 @@ def _test_flip(self, use_cuda=False):
     def test_flip(self):
         self._test_flip(self, use_cuda=False)
 
+    def test_reversed(self):
+        val = torch.arange(1, 10)
```
Comment (SsnL) [inline test/test_torch.py]: but this should be `(0, 10)`. :D

Comment (vishwakftw) [reply test/test_torch.py]: Oh no. My bad again. Sorry.

**Comment by vishwakftw** [comment]
Is this good to go? Sorry about the reminder.

**Review by apaszke — APPROVED** [review]

**Pull request closed without merging**